backend/app/chat/routers.py

```diff
@@ -31,14 +35,17 @@ async def chat(
     and does not retrieve chat history.
     """
 
+    chat_request = await update_request_using_history(chat_request, asession)
```

Comment:
@sidravi1 - just want to check whether these should adhere to service-based approach. @robela do you want these functions stored elsewhere?

Reply:
Yes! I will move all of these to a separate module / class